### ISSUE TITLE
[BUG] Predict depends on memory layout

### ISSUE THREAD

**Issue by MichaelMarien** [jmschrei/pomegranate#717]
First of all, thanks a lot for the library, it's great. I ran into following bug, I'm happy to help fix it!

**Describe the bug**
Fitted models report different prediction results depending on the memory layout of the numpy array. The user should be shielded from such behaviour as this is a low level detail that is not clear from the numpy API. 

I got this in a real live use-case by getting my numpy array to predict on from a doubly indexed pandas dataframe. I assume pandas stores it row-wise instead of column wise for optimised index lookups (not sure about this).

**To Reproduce**

```
from pomegranate import *
import numpy as np

data = np.array([[1, 0], [0, 1], [1, 0], [1, 1] ,[1, 0]])
data_other = np.copy(data, order="F")

model = GeneralMixtureModel.from_samples([BernoulliDistribution, BernoulliDistribution], n_components=2, X=data)

model.predict_proba(data) == model.predict_proba(data_other)
```

**Comment by jmschrei** [comment]
Yes, this is absolutely an issue and I agree that users should be shielded from it. Essentially, data matrices are assumed to be C styled when they reach the Cython part of the code even when the numpy array has the F flag. This can be solved by adding in a check for each function to make sure that the order is C and that the strides tuple is ordered correctly. If you can code such a check I can add it to each function.

**Comment by MichaelMarien** [comment]
what about including this in the utils. _check_input? Currently it explicitly converts X to an array, adding an order parameter these is easy. This would mean we need to include this _check_input everywhere (e.g. probability methods of distributions). I'm a little worried about missing some spots. Let me know what you think. I can definitely add some tests also.

**Comment by MichaelMarien** [comment]
I'll close this as at least a lot of common cases should be fixed by #740. If other users experience a similar issue, they can reopen it.
